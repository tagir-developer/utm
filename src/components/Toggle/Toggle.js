import React from 'react'
import classes from './Toggle.module.css'

const Toggle = ({themeToggle, darkTheme}) => {

	return (
		<div className={classes.main}>
			<input 
				type="checkbox" 
				id={classes.cbx} 
				style={{ display: 'none' }} 
				onChange={themeToggle}
				checked={darkTheme}
				special="test"
			/>
			<label htmlFor={classes.cbx} className={classes.toggle}><span></span></label>
		</div>
	)
}

Toggle.defaultProps = {
	themeToggle: () => {},
	darkTheme: false,
}

export default Toggle