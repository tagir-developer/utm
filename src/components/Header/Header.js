import React from 'react'
import Toggle from '../Toggle/Toggle'

const Header = props => {

	const bgImage = {
		backgroundImage: 'url(./images/header-4.jpg)',
		backgroundPosition: 'right',
		height: '200px'
	}

	return (
		<div className="text-light p-4 mb-5 text-right" style={bgImage}>
			<div className="d-none d-sm-block m-2"></div>
			<h1 className="display-4 d-none d-md-block">Генератор <wbr/><span className="text-info">UTM</span>&#8209;меток</h1>
			{/* Heading for mobile devices */}
			<h1 className="display-5 d-block d-md-none">Генератор <wbr/><span className="text-info">UTM</span>&#8209;меток</h1>
			<div className="d-flex flex-row justify-content-end mt-4">
				<div className="font-weight-bolder mr-3">{props.darkTheme ? 'Выключить ночной режим' : 'Включить ночной режим'}</div>
				<Toggle themeToggle={props.themeToggle} darkTheme={props.darkTheme} />
			</div>
		</div>
	)
}

Header.defaultProps = {
	themeToggle: () => {},
	darkTheme: false,
}

export default Header