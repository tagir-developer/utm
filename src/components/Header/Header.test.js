import Header from './Header'
import Toggle from '../Toggle/Toggle'
import React from 'react'

describe('<Header/>', () => {

	let wrapper

	beforeEach(() => {
		wrapper = shallow(<Header />)
	})

	describe('Header render', () => {

		it('Should Header component render', () => {
			expect(toJson(wrapper)).toMatchSnapshot()
		})

		it('Should include toggle component', () => {
			expect(wrapper.find(Toggle)).toHaveLength(1)
		})

		it('If app theme is dark - should render "Выключить ночной режим"', () => {
			wrapper.setProps({
				darkTheme: true
			})
			expect(wrapper.find('div.font-weight-bolder.mr-3').text()).toEqual('Выключить ночной режим')
		})

		it('If app theme is default (white) - should render "Включить ночной режим"', () => {
			wrapper.setProps({
				darkTheme: false
			})
			expect(wrapper.find('div.font-weight-bolder.mr-3').text()).toEqual('Включить ночной режим')
		})

	})

	describe('Default props', () => {

		it('Should use default themeToggle', () => {
			const result = Header.defaultProps.themeToggle()
			expect(result).toBe(undefined)
		})

		it('With default prop darkTheme should render "Включить ночной режим"', () => {
			expect(wrapper.find('div.font-weight-bolder.mr-3').text()).toEqual('Включить ночной режим')
		})

	})

})


