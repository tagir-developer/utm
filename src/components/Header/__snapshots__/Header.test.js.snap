// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Header render Should Header component render 1`] = `
<div
  className="text-light p-4 mb-5 text-right"
  style={
    Object {
      "backgroundImage": "url(./images/header-4.jpg)",
      "backgroundPosition": "right",
      "height": "200px",
    }
  }
>
  <div
    className="d-none d-sm-block m-2"
  />
  <h1
    className="display-4 d-none d-md-block"
  >
    Генератор 
    <wbr />
    <span
      className="text-info"
    >
      UTM
    </span>
    ‑меток
  </h1>
  <h1
    className="display-5 d-block d-md-none"
  >
    Генератор 
    <wbr />
    <span
      className="text-info"
    >
      UTM
    </span>
    ‑меток
  </h1>
  <div
    className="d-flex flex-row justify-content-end mt-4"
  >
    <div
      className="font-weight-bolder mr-3"
    >
      Включить ночной режим
    </div>
    <Toggle
      darkTheme={false}
      themeToggle={[Function]}
    />
  </div>
</div>
`;
